src/main/java/htsjdk/variant/variantcontext/VariantContext.java

```diff
@@ -1353,9 +1353,9 @@ private void validateAlleles() {
     private void validateGenotypes() {
         if ( this.genotypes == null ) throw new IllegalStateException("Genotypes is null");
 
-        for ( final Genotype g : this.genotypes ) {
-            if ( g.isAvailable() ) {
-                for ( Allele gAllele : g.getAlleles() ) {
+        for ( int i = 0; i < genotypes.size(); i++ ) {
+            if ( genotypes.get(i).isAvailable() ) {
```

Comment:
pull out the get(i) as a variable if speed is so essential here.

Reply:
shouldn't the inner loop be made index because that will be the most essential one

Reply:
Yes, I missed that. Though its worth mentioning that most of the improvement in this method came from genotypes.isAvailible which was more expensive than the inner loop.  I'll extract it out however since we are here

Reply:
I would say this is probably the grosses of the unreadable code after this change...